classes/factorial.rb

```diff
@@ -0,0 +1,15 @@
+# The Factorial class provides methods to calculate the factorial of a number.
+class Factorial
+  # Calculates the factorial of a given number.
+  def factorial(num)
+    return 1 if num.zero?
+
+    num * factorial(num - 1)
+  end
+
+  def find(num)
+    raise ArgumentError, 'Argument must be greater than or equal to 0.' if num.negative?
+
+    factorial(num)
+  end
+end
```

Comment:
- [ ] The requirements state that you should create a `Solver` class, with `factorial`, `reverse` and `fizzbuzz` methods that each take one argument. Then implement the logic for each method based on the requirements as shown in this [article](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md). 👍 😎 

Please create _**one**_ file that contains the `Solver` class with the three methods. 💻 

#### Hint: You can follow the steps below:

1. Create a `solver.rb` file with an empty `Solver` class and `git commit`.
2. create a test file for the `Solver` class and `git commit`.
3. Implement TDD for the `factorial` method of the `Solver` class in the test file and `git commit`.
4. Write logic for the `factorial` method in the main `Solver` class and `git commit`.
5. Implement TDD for the `reverse` method of the solver class in the test file and `git commit`.
6. Write logic for the `reverse` method in the main `Solver` class and `git commit`.
7. Implement TDD for the `fizzbuzz` method of the solver class in the test file and `git commit`.
8. Write logic for the `fizzbuzz` method in the main `Solver` class and `git commit`. 🖱️

Reply:
Hello @Gambit142.
Unfortunately, It is impossible for us to place all methods in one class since there's a linter error that shows the class holding all three methods `is too long`. 

Here is the error: 
spec/solver.rb:6:1: C: Metrics/BlockLength: Block has too many lines. [41/25]
describe Solver do ...
^^^^^^^^^^^^^^^^^^^

We feel that it is a bit impractical. Our fellow Micronauts from morning and evening standups have had their projects approved while having written the methods in separate files instead of one Class. Does this mean we still have to follow the instructions regarding writing all methods under one class?